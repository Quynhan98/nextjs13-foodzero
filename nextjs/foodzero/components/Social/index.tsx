import React, { memo } from 'react'
import Image, { ImageProps } from 'next/image'
import Link from 'next/link'

interface SocialProps extends ImageProps {
  link: string
}

const Social = ({ link, width = 28, height = 28, ...rest }: SocialProps) => {
  return (
    <Link href={link} prefetch={false}>
      <Image width={width} height={height} {...rest} />
    </Link>
  )
}

export default memo(Social)
