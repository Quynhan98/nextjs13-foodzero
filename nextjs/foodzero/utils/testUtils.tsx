import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
