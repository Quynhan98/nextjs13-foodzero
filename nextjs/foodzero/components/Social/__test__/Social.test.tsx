import React from 'react'

// component
import Social from '@components/Social'

// Utils
import { render, screen } from '@utils/testUtils'

describe('Social render', () => {
  const props = {
    alt: 'facebook',
    src: '/images/facebook.webp',
    link: '/',
  }

  it('Should show match Social DOM Snapshot', () => {
    const { container } = render(<Social {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render correct link', () => {
    render(<Social {...props} />)

    const link = screen.getByRole('link')
    expect(link.getAttribute('href')).toEqual(props.link)
  })
})
