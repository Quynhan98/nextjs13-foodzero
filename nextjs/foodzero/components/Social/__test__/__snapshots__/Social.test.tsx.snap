// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social render Should show match Social DOM Snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="facebook"
      data-nimg="1"
      decoding="async"
      height="28"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Ffacebook.webp&w=64&q=75"
      srcset="/_next/image?url=%2Fimages%2Ffacebook.webp&w=32&q=75 1x, /_next/image?url=%2Fimages%2Ffacebook.webp&w=64&q=75 2x"
      style="color: transparent;"
      width="28"
    />
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
